File: backend/auth-service/src/main/java/com/bikerental/auth_service/entity/DepositDeduction.java
package com.bikerental.auth_service.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "deposit_deduction")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DepositDeduction {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "deduction_id")
	private Integer deductionId;

	@ManyToOne
	@JoinColumn(name = "booking_id", nullable = false)
	private BikeBookingDetails booking;

	@Column(nullable = false)
	private String description;

	@Column(nullable = false, precision = 10, scale = 2)
	private BigDecimal amount;

	@Column(name = "document_url")
	private String documentUrl;

	@ManyToOne
	@JoinColumn(name = "recorded_by")
	private User recordedBy;

	@Column(name = "created_at", nullable = false)
	private LocalDateTime createdAt;

	@Column(nullable = false)
	private String status;

	@Column(name = "disputed_at")
	private LocalDateTime disputedAt;

	@Column(name = "dispute_reason")
	private String disputeReason;

	@Column(name = "resolved_at")
	private LocalDateTime resolvedAt;

	@Column(name = "resolution_note")
	private String resolutionNote;

	@ManyToOne
	@JoinColumn(name = "resolved_by")
	private User resolvedBy;
}

File: backend/booking-service/Domain/Entities/DepositDeduction.cs
using Booking.Api.Domain.Enums;

namespace Booking.Api.Domain.Entities;

public class DepositDeduction
{
    public int DeductionId { get; set; }        // deduction_id (serial PK)
    public int BookingId { get; set; }          // FK -> bike_booking_details (kept)

    public string Description { get; set; } = default!;
    public decimal Amount { get; set; }
    public string? DocumentUrl { get; set; }

    public int? RecordedBy { get; set; }        // user_id in auth-service (no FK)
    public DateTime CreatedAt { get; set; }

    public DeductionStatus Status { get; set; }

    // 48-hour dispute window fields
    public DateTime? DisputedAt { get; set; }
    public string? DisputeReason { get; set; }
    public DateTime? ResolvedAt { get; set; }
    public string? ResolutionNote { get; set; }
    public int? ResolvedBy { get; set; }        // user_id in auth-service (no FK)

    public BikeBooking Booking { get; set; } = default!;   // navigation
}